src/helpers/createProof.ts

```diff
@@ -20,6 +21,10 @@ export default async function (
         'zk/AllowMapChecker.wasm',
         'zk/AllowMapChecker_final.zkey'
       )
+
+      if (await checkIfProofUsedBefore(type, proof.publicSignals[1])) {
+        return Promise.reject(new Error(`This token has already been used!`))
```

Comment:
What if we use `throw new Error` here, instead of `Promise.reject`?
Otherwise looks good